ledger/src/blockstore.rs

```diff
@@ -1452,14 +1504,99 @@ impl Blockstore {
         slot: Slot,
         index: u64,
     ) -> Cow<'a, Vec<u8>> {
-        let key = ShredId::new(slot, u32::try_from(index).unwrap(), ShredType::Data);
+        self.get_shred_from_just_inserted_or_db(
+            just_inserted_shreds,
+            slot,
+            u32::try_from(index).unwrap(),
```

Comment:
this is annoying, I've tracked down the culprit to `SlotMeta`, which stores indices `consumed` and `received` as `u64`.